File: src/main/java/metaside/MiddleEarthCitizen.java
package metaside;

import java.util.Random;

/**
 * Created by Petr on 15.10.2016.
 */
public abstract class MiddleEarthCitizen {
    private int power;
    private boolean dead;

    public MiddleEarthCitizen(int minPower, int maxPower){
        Random rand = new Random(System.currentTimeMillis());
        this.power = BattleRandom.getInstance().intBetween(minPower, maxPower);
        this.dead = false;
    }

    public MiddleEarthCitizen(int power) {
        this.power = power;
    }

    public void applyAttack(MiddleEarthCitizen enemy){
        enemy.setPower(enemy.getPower() - getPower());
    }

    public int getPower() {
        return power;
    }

    public void setPower(int power) {
        this.power = power;
        this.setDead(this.power <= 0);
    }

    public String toString() {
        return this.getClass().getName();
    }

    public boolean isDead() {
        return dead;
    }

    public void setDead(boolean dead) {
        this.dead = dead;
    }
}

File: CSharpVersion/MiddleEarth/MiddleEarth/metaside/MiddleEarthCitizen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MiddleEarth.metaside
{
    class MiddleEarthCitizen
    {
        private int power;
        private bool dead;
        private string name;

        public MiddleEarthCitizen(string name, int minPower, int maxPower)
        {
            this.name = name;

            Random rand = new Random(Guid.NewGuid().GetHashCode());
            this.power = rand.Next(minPower, maxPower);
            this.dead = false;
        }

        public MiddleEarthCitizen(string name, int power)
        {
            this.name = name;
            this.power = power;
        }

        public void applyAttack(MiddleEarthCitizen enemy)
        {
            enemy.setPower(enemy.getPower() - getPower());
        }

        public string getName()
        {
            return this.name;
        }

        public int getPower()
        {
            return power;
        }

        public void setPower(int power)
        {
            this.power = power;
            this.setDead(this.power <= 0);
        }

        public bool isDead()
        {
            return dead;
        }

        public void setDead(bool dead)
        {
            this.dead = dead;
        }
    }
}
